browser-switch/src/main/java/com/braintreepayments/api/BrowserSwitchObserver.java

```diff
@@ -0,0 +1,93 @@
+package com.braintreepayments.api;
+
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.VisibleForTesting;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentActivity;
+import androidx.fragment.app.FragmentManager;
+
+import java.util.List;
+
+public class BrowserSwitchObserver {
+
+    private final BrowserSwitchPersistentStore persistentStore;
+    private final BrowserSwitchListenerFinder listenerFinder;
+
+    public BrowserSwitchObserver() {
+        this(BrowserSwitchPersistentStore.getInstance(), new BrowserSwitchListenerFinder());
+    }
+
+    @VisibleForTesting
+    BrowserSwitchObserver(BrowserSwitchPersistentStore persistentStore, BrowserSwitchListenerFinder listenerFinder) {
+        this.persistentStore = persistentStore;
+        this.listenerFinder = listenerFinder;
+    }
+
+    public void onActivityResumed(FragmentActivity activity) {
+        BrowserSwitchResult result = getResult(activity);
+        if (result != null) {
+
+            Context appContext = activity.getApplicationContext();
+            BrowserSwitchRequest request = persistentStore.getActiveRequest(appContext);
+
+            List<BrowserSwitchListener> listeners = listenerFinder.findActiveListeners(activity);
+            for (BrowserSwitchListener listener : listeners) {
+                listener.onBrowserSwitchResult(result);
+            }
```

Comment:
Maybe this should all be consolidated - could we call `onBrowserSwitchResult` on the listeners as we are finding them? Might be unnecessary to loop through the list twice.

Reply:
It could be considered unnecessary but I think the tradeoff here makes `BrowserSwitchObserver` much easier to test since we don't have to think about `FragmentManager` or `instanceof`.

Reply:
In the future, we could convert `BrowserSwitchListenerFinder` into an interface and let merchants define their own strategy for what's considered a listener, and have a `DefaultBrowserSwitchListenerFinder` implementation that will work for standard integrations.